api/v1beta1/cryostat_types.go

```diff
@@ -88,6 +88,21 @@ type CryostatSpec struct {
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Credentials Database Options"
 	JmxCredentialsDatabaseOptions *JmxCredentialsDatabaseOptions `json:"jmxCredentialsDatabaseOptions,omitempty"`
+	// Options to configure the Cryostat resources metadata
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Resources metadata"
+	Metadata *ResourceMetadata `json:"metadata,omitempty"`
```

Comment:
I'd like to differentiate this from other metadata defined elsewhere in this CRD (e.g. services, ingress). Maybe `PodMetadata` or `WorkloadMetadata`, any preferences @andrewazores?

Reply:
`PodMetadata` or even `OperandMetadata` maybe, since that name would also suggest that the metadata could be applied to ex. the `Deployment`?

Reply:
My initial idea was to provide there labels and annotations that would apply to all resources created, but you're right I missed the ingress/services :smile: 

Should I split that in 2 to differentiate the deployment and pod metadata, or apply it on top for the ingress/services as well ?

Reply:
I suppose if it's possible to configure custom labels/annotations like this then it may as well be fully customizable for each kind of resource. On the other hand, I imagine that most of the time users will just want the same labels/annotations across all of the resources, so duplicating the configuration for each kind is a bit of a waste. Perhaps in that case the user can put their common labels/annotations into a ConfigMap and reference that from each resource kind's custom metadata config...

@OpenGuidou which way would make more sense to you as a user? It's easy for me to look at this and think that if it's done at all then it should be done with maximum flexibility, but that also entails maximum complexity :-)

Reply:
For my I would look at it the way we are using helm templates usually: have common labels/annotations that apply everywhere, then let the possibility to override it at each level (resource type).
Then it'll complexify a bit the code here :smile:

Reply:
Well after a bit of thinking, I guess the lesser of two evils would indeed to differentiate pod and deployment metadata in the configuration. Not a lot of overload in the configuration, quite simple in the code and we keep the flexibility.

Are you ok with that ?

Reply:
Sounds fine to me. Would that also have separate metadata configuration for ingress/services too? And each of the pod/deployments for the main Cryostat container, the reports sidecar(s), the jfr-datasource, and the Grafana dashboard...

Reply:
Updated

Reply:
Sounds good to me.